### DISCUSSION TITLE
Why doesn't getRelationship() allow hasMany on a Field relationship()?

### DISCUSSION THREAD

**Discussion by joshembling** [filamentphp/filament#6700] [General]
Currently, `getRelationship()` only allows BelongsTo, BelongsToMany, BelongsToThrough, or null

Which means my current relationships between models `Device` and `Image` cannot work when I chain `relationship()` on the Device Resource, but I would like to know why this is the case? It is a pretty basic relationship that I think should work out of the box by default.

```
// Device.php
public function images()
{
    return $this->hasMany(Image::class);
}
```

```
// Image.php
public function device()
{
    return $this->belongsTo(Device::class);
}
```

```
// Migration
Schema::create('images', function (Blueprint $table) {
    $table->id();
    $table->unsignedBigInteger('device_id');
    $table->string('filename');
    $table->timestamps();
});
```
```
// DeviceResource.php
public static function form(Form $form): Form
{
    return $form
        ->schema([
            Fieldset::make('Images')
                ->relationship('images')
                ->schema([
                          FileUpload::make('filename')
                ]),
```

Returns the following error msg
`Filament\Forms\Components\Fieldset::getRelationship(): Return value must be of type Illuminate\Database\Eloquent\Relations\BelongsTo|Illuminate\Database\Eloquent\Relations\HasOne|Illuminate\Database\Eloquent\Relations\MorphOne|null, Illuminate\Database\Eloquent\Relations\HasMany returned`

**Comment by danharrin** [comment]
Because you want a Repeater for this, not a layout component which saves to a singular relationship
